### PR TITLE
fix: login error method

### PR DESCRIPTION
**Pull request by noneAuth**
Fix the problem of unclear logic in the request interceptor method, which leads to slower execution of synchronous code messages than jumps.

### CODE REVIEW

File: client/src/components/studio/utils/request.tsx [hunk 1]
```diff
@@ -17,14 +17,27 @@ const request = extend({
...
+  const TUGRAPH_TOKEN = getLocalData('TUGRAPH_TOKEN');
   if (data.errorCode == 401 || data.errorCode == 400) {
-    message.warning('登录过期，请重新登录');
-    // 清除掉之前的缓存
-    localStorage.removeItem('TUGRAPH_TOKEN');
-    window.location.href = '/login';
+    new Promise<void>(resolve => {
+      setTimeout(() => {
+        if (
+          (!TUGRAPH_TOKEN && data.errorCode == 400) ||
```
Comment (lxfu1) [inline client/src/components/studio/utils/request.tsx]: 重复判断

File: client/src/components/studio/utils/request.tsx [hunk 2]
```diff
@@ -17,14 +17,27 @@ const request = extend({
...
+          data.errorCode == 401
+        ) {
+          message.warning('登录过期，请重新登录', 2.5);
```
Comment (lxfu1) [inline client/src/components/studio/utils/request.tsx]: 这个应该在外层判断吧，不应该在 promise 内部

File: client/src/components/studio/login/index.tsx [hunk 3]
```diff
@@ -32,26 +32,31 @@ export const Login = () => {
           if (res.errorCode == 200) {
             setLocalData('TUGRAPH_USER_NAME', values.userName);
             setLocalData('TUGRAPH_TOKEN', res.data.authorization);
-            message.success('登录成功！');
+            message.success('登录成功！', 2.5);
```
Comment (lxfu1) [inline client/src/components/studio/login/index.tsx]: 2.5改为通用的 constants

Comment (noneAuth) [reply client/src/components/studio/login/index.tsx]: done

File: client/src/components/studio/utils/request.tsx (hunk 2 above)
Comment (noneAuth) [reply client/src/components/studio/utils/request.tsx]: done

File: client/src/components/studio/utils/request.tsx (hunk 1 above)
Comment (noneAuth) [reply client/src/components/studio/utils/request.tsx]: done

**Review by lxfu1 — APPROVED** [review]

**Pull request merged**